### PR TITLE
fix: isolate doc ingestion and query into Processes

### PR DESCRIPTION
**Pull request by marcelklehr**

### CODE REVIEW

File: context_chat_backend/chain/ingest/injest.py [hunk 1]
```diff
@@ -122,17 +122,16 @@ def _process_sources(
...
+) -> None:
 	filtered_sources = _filter_sources(sources[0].get('userId'), vectordb, sources)
 
 	if len(filtered_sources) == 0:
 		# no new sources to embed
 		print('Filtered all sources, nothing to embed', flush=True)
-		result['success'].set()
 		result['done'].set()
-		return True
+		result['success'].set()
```
Comment (marcelklehr) [inline context_chat_backend/chain/ingest/injest.py:134]: This is the wrong way around, I think. Controller code waits for done. If success is set after done, we have a race condition.

File: context_chat_backend/chain/ingest/injest.py [hunk 2]
```diff
@@ -122,17 +122,16 @@ def _process_sources(
...
-):
+) -> None:
 	filtered_sources = _filter_sources(sources[0].get('userId'), vectordb, sources)
 
 	if len(filtered_sources) == 0:
 		# no new sources to embed
 		print('Filtered all sources, nothing to embed', flush=True)
-		result['success'].set()
 		result['done'].set()
-		return True
```
Comment (marcelklehr) [inline context_chat_backend/chain/ingest/injest.py:122]: Why remove the returns?

File: context_chat_backend/chain/ingest/injest.py (hunk 1 above)
Comment (kyteinsky) [reply context_chat_backend/chain/ingest/injest.py:134]: ah you're right, this must be the deadlock in the integration test

File: context_chat_backend/chain/ingest/injest.py (hunk 2 above)
Comment (kyteinsky) [reply context_chat_backend/chain/ingest/injest.py:122]: opps, don't know what I was thinking

**Comment by kyteinsky** [comment]
maybe the embedding took more than 30 mins?

File: context_chat_backend/workers/embedding.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        print('[embedding_worker] Waiting for vectordb_lock')
+        with vectordb_lock:
+            print('[embedding_worker] Got vectordb_lock, adding documents to vectordb')
+            try:
+                user_client = db.get_user_client(user_id)
+                count = len(user_client.add_documents(split_documents))
+            except Exception as e:
+                print(e)
+                count = 0
```
Comment (marcelklehr) [inline context_chat_backend/workers/embedding.py:33]: Ah, my bad, this is not good i guess :sweat_smile:  #variableclash

Comment (kyteinsky) [reply context_chat_backend/workers/embedding.py:33]: sorry didn't get it, which variable clashes?

Comment (marcelklehr) [reply context_chat_backend/workers/embedding.py:33]: the loop counter is also called `count`

**Comment by marcelklehr** [comment]
This PR is too slow.

**Pull request closed without merging**